core-api/src/main/java/org/testng/internal/objects/InstanceCreator.java

```diff
@@ -21,12 +21,15 @@ private InstanceCreator() {
 
   public static <T> T newInstance(Class<T> clazz) {
     try {
-      return clazz.newInstance();
+      return clazz.getDeclaredConstructor().newInstance();
```

Comment:
Ok with the change, but why?

Reply:
`clazz.newInstance();` is deprecated in Java9+, and `build-logic` has `-Xlint:deprecation + -Werror` which implies "fail when deprecated APIs are used".